src/main/java/mate/academy/dao/CinemaHallDao.java

```diff
@@ -0,0 +1,8 @@
+package mate.academy.dao;
+
+import java.util.List;
+import mate.academy.model.CinemaHall;
+
+public interface CinemaHallDao extends GenericDao<CinemaHall> {
+    List<CinemaHall> getAll();

```

Comment:
If you introduced GenericDao why getAll() method is not included there but separately hardcoded in each dao?

Reply:
I have three dao interfaces that extends GenericDao, and according to requirements only two of them have getAll method. If i'll add getAll() method to GenericDao, then I need to implement it in MovieSessionDao, but README.md tells me that it should have only add, get and findAvailableSessions methods. In a list with common mistakes I have recomendation "Do not create redundant functionality, like methods that are not required by the task." So, that's why I did as I did :) The same with Service layer.